File: src/Application/ClientApp/app/models/order.model.ts
import { OrderHistoryModel } from './order-history.model';
import { OrderDetailsModel } from './order-details.model';
import { UserModel } from './user-model';

export class OrderModel {
    public id: string;
    public orderNumber: string;
    public countOfWares: number;
    public createdDate: Date;
    public status: string;
    public deliveryService: string;
    public declarationNumber: string;
    public orderHistories: OrderHistoryModel[];
    public user: UserModel;
    public orderDetails: OrderDetailsModel[];
}

File: src/Entities/Models/OrderModels/OrderModel.cs
using System;
using System.Collections.Generic;

namespace Application.EntitiesModels.Models.OrderModels
{
    public class OrderModel
    {
        public Guid Id { get; set; }
        public string OrderNumber { get; set; }
        public int CountOfWares { get; set; }
        public DateTime CreatedDate { get; set; }     
        public string Status { get; set; }
        public string DeliveryService { get; set; }
        public string DeclarationNumber { get; set; }

        public UserModel User { get; set; }
        public List<OrderHistoryModel> OrderHistories { get; set; }
        public List<OrderDetailsModel> OrderDetails { get; set; }

        public OrderModel()
        {
            OrderHistories = new List<OrderHistoryModel>();
            OrderDetails = new List<OrderDetailsModel>();
        }
    }
}
